src/GraphQL.MicrosoftDI/AutoCreatingServiceProvider.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// An <see cref="IServiceProvider"/> wrapper which instantiates classes that are unregistered in the
+    /// underlying service provider. Intended to be passed to the <see cref="Schema"/> constructor so that
+    /// all of the graph types do not individually need to be registered within your DI container.
+    /// <br/><br/>
+    /// To use this, simply register the <see cref="ISchema"/> itself as follows:
+    /// <br/><br/>
+    /// <code>services.AddSingleton&lt;ISchema, MySchema&gt;(services => new MySchema(new AutoCreatingServiceProvider(services)));</code>
+    /// <br/><br/>
+    /// Within your <see cref="Schema"/> constructor, you may need to set your <see cref="Schema.Query"/>, <see cref="Schema.Mutation"/> and
+    /// <see cref="Schema.Subscription"/> fields to pull from the <see cref="AutoCreatingServiceProvider"/>:
+    /// <br/><br/>
+    /// <code>Query = services.GetRequiredService&lt;MyQuery&gt;();</code>
+    /// <br/><br/>
+    /// None of the graph types will need to be registered.
+    /// </summary>
+    public class AutoCreatingServiceProvider : IServiceProvider
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        /// <summary>
+        /// Creates a new instance with the specified underlying service provider.
+        /// </summary>
+        public AutoCreatingServiceProvider(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+        }
+
+        /// <inheritdoc cref="ActivatorUtilities.GetServiceOrCreateInstance(IServiceProvider, Type)"/>
+        public object GetService(Type serviceType) => ActivatorUtilities.GetServiceOrCreateInstance(_serviceProvider, serviceType);
```

Comment:
From time to time I meet this method. I understand what it is doing. Its name speaks for itself :). But I was always interested in another question - what happens in the event of creating `IDisposable` instances. I got the impression that they were not disposed. Now I once again looked at the source code and my assumption seems true, although I could miss something 🤔 . Please write a test with this provider, which checks the fact of calling the `Dispose` method when destroying the container. If this method calls, it means that I carefully study the source code. Otherwise, it will be necessary to clearly note in the documentation that the types unregistered in the container should not implement the `IDisposable` interface (more precisely, they can do it, but the method will not be called).

Reply:
Yep. I could not resist and did the test.
```csharp
    public class Repository : IDisposable
        {
            public Repository()

            {

            }
            public void Dispose()
            {
                Console.WriteLine("Called");
            }
        }

        public class My : IDisposable
        {
            public My(Repository r)
            {

            }

            public void Dispose()
            {
                Console.WriteLine("Not called");
            }
        }

 [Fact]
        public void aaa()
        {
            var services = new ServiceCollection();
            services.AddSingleton(typeof(Repository));
            var provider = services.BuildServiceProvider();
            var inst = ActivatorUtilities.GetServiceOrCreateInstance(provider, typeof(My));

            provider.Dispose();
```

> there's no real reason to have to register each type within your service provider. 

**IF** this type does not implement `IDisposable`. This is a weighty remark.

Reply:
Yeah; all the dependencies will be properly disposed.

Reply:
I added documentation.  I don't understand what kind of test you would add.  One that demonstrates that desired behavior does NOT occur? There no failure condition.  Either it does not occur, which we know about, or it does which we would want. I would not add a `[Fact(Skip = true)]` test either, because this is expected behavior.

Reply:
> One that demonstrates that desired behavior does NOT occur?

Yes. This must be at least in order to protect against unintentional changes caused by either by our code or by .NET runtime code.